packages/xsnap/test/test-xs-js.js

```diff
@@ -50,6 +51,61 @@ test('simple TextEncoder / TextDecoder are available', async t => {
   t.deepEqual(opts.messages, ['"Hello! 😊"', '"A\\u0000A"']);
 });
 
+test('Base64.encode', async t => {
+  const opts = options(io);
+  const vat = xsnap(opts);
+  t.teardown(() => vat.terminate());
+  await vat.evaluate(`
+    const encoder = new TextEncoder();
+    globalThis.handleCommand = inputBuffer => {
+      const outputString = Base64.encode(inputBuffer);
+      const outputUint8Array = encoder.encode(outputString);
+      globalThis.issueCommand(outputUint8Array.buffer);
+    };
+  `);
+  const inputUint8Array = new TextEncoder().encode('Hello, World! 😃🌍');
+  const expectedOutputString = encodeBase64(inputUint8Array);
+  await vat.issueCommand(inputUint8Array);
+  t.deepEqual(opts.messages, [expectedOutputString]);
+});
+
+test.failing('Base64.encode degenerate input case', async t => {
+  const opts = options(io);
+  const vat = xsnap(opts);
+  t.teardown(() => vat.terminate());
+  await vat.evaluate(`
+    const encoder = new TextEncoder();
+    globalThis.handleCommand = inputBuffer => {
+      const outputString = Base64.encode(inputBuffer);
+      const outputUint8Array = encoder.encode(outputString);
+      globalThis.issueCommand(outputUint8Array.buffer);
+    };
+  `);
+  const inputUint8Array = new Uint8Array();
+  const expectedOutputString = '';
+  await vat.issueCommand(inputUint8Array);
+  t.deepEqual(opts.messages, [expectedOutputString]);
+});
+
+test('Base64.decode', async t => {
```

Comment:
This file was intended to be for JS correctness tests. I wonder about adding XS extensions here.

Hang on... the web convention seems to be [atob](https://developer.mozilla.org/en-US/docs/Web/API/atob) and [btoa](https://developer.mozilla.org/en-US/docs/Web/API/btoa). Did you consider that API?

Reply:
Yes. btoa/atob are quite old and problematic. Their domain/range is ASCII strings.

Reply:
Ah. I see. Thanks.